src/gui_common/CellHexPhotoBuilder.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using Godot;
+
+public class CellHexPhotoBuilder : Spatial, IPhotographable
+{
+    private float radius;
+    private bool radiusDirty;
+    private MicrobeSpecies? species;
+
+    public string SceneToPhotographPath => "res://src/gui_common/CellHexPhotoBuilder.tscn";
+
+    public float Radius
+    {
+        get
+        {
+            if (radiusDirty)
+                CalculateRadius();
+
+            return radius;
+        }
+    }
+
+    public MicrobeSpecies? Species
+    {
+        get => species;
+        set
+        {
+            species = value;
+            radiusDirty = true;
+        }
+    }
+
+    public void ApplySceneParameters(Spatial instancedScene)
+    {
+        var builder = (CellHexPhotoBuilder)instancedScene;
+        builder.Species = Species;
+        builder.BuildHexStruct();
+    }
+
+    public float CalculatePhotographDistance(Spatial instancedScene)
+    {
+        return PhotoStudio.CameraDistanceFromRadiusOfObject(((CellHexPhotoBuilder)instancedScene).Radius *
+            Constants.PHOTO_STUDIO_CELL_RADIUS_MULTIPLIER);
+    }
+
+    private void BuildHexStruct()
+    {
+        var hexScene = GD.Load<PackedScene>("res://src/microbe_stage/editor/EditorHex.tscn");
+        var validMaterial = GD.Load<Material>("res://src/microbe_stage/editor/ValidHex.material");
+        var modelScene = GD.Load<PackedScene>("res://src/general/SceneDisplayer.tscn");
+
+        foreach (Node node in GetChildren())
+            node.DetachAndQueueFree();
+
+        if (Species == null)
+            throw new InvalidOperationException("Species is ont initialized");
+
+        var organelleLayout = Species.Organelles;
+        foreach (var organelle in organelleLayout.Organelles)
+        {
+            var position = organelle.Position;
+            var itemHexes = organelle.RotatedHexes;
+            foreach (var hex in itemHexes)
+            {
+                var pos = Hex.AxialToCartesian(hex + position);
+
+                var hexNode = (MeshInstance)hexScene.Instance();
+                AddChild(hexNode);
+                hexNode.MaterialOverride = validMaterial;
+                hexNode.Translation = pos;
+            }
+        }
+
+        foreach (var organelle in organelleLayout)
+        {
+            // Hexes are handled by UpdateAlreadyPlacedHexes
```

Comment:
This seems like copied code from the hex editor...

I'd really like to avoid this much logic duplication because changing this in the future will be harder.
I think a few static methods in the organelle editor that do stuff like instance and place the scenes in the right places would be good so that this and the editor could share the same core logic, which is complex code.

Reply:
They are somehow different because this builder simplifies the logic greatly. I cannot merge them together...

Reply:
I was thinking the general logic would work like so that you give it a single organelle, along with callbacks it can use to ask for things like the graphics holders. I think while that would be a bit complicated to make, it would still allow 80-90% code sharing, and all of the complex math could be shared.